kedro-datasets/kedro_datasets/netcdf/netcdf_dataset.py

```diff
@@ -0,0 +1,218 @@
+        if self._protocol != "file":
+            logger.info("Syncing remote NetCDF file to local storage.")
+
+            # `get_filepath_str` drops remote protocol prefix.
+            load_path = self._protocol + "://" + load_path
+            if self._is_multifile:
+                load_path = sorted(self._fs.glob(load_path))
+
+            self._fs.get(load_path, f"{self._temppath}/")
+            load_path = f"{self._temppath}/{self._filepath.stem}.nc"
+
+        if "*" in str(load_path):
+            data = xr.open_mfdataset(str(load_path), **self._load_args)
+        else:
+            data = xr.open_dataset(load_path, **self._load_args)
+
+        return data
+
+    def _save(self, data: xr.Dataset):
+        if self._is_multifile:
+            raise DatasetError(
+                "Globbed multifile datasets with '*' in filepath cannot be saved. "
+                + "Create an alternate NetCDFDataset with a single .nc output file."
+            )
+        else:
+            save_path = get_filepath_str(self._filepath, self._protocol)
+
+            if self._protocol != "file":
+                # `get_filepath_str` drops remote protocol prefix.
+                save_path = self._protocol + "://" + save_path
+
+            bytes_buffer = data.to_netcdf(**self._save_args)
+
+            with self._fs.open(save_path, mode="wb") as fs_file:
+                fs_file.write(bytes_buffer)
+
+            self._invalidate_cache()
+
+    def _describe(self) -> Dict[str, Any]:
+        return dict(
+            filepath=self._filepath,
+            protocol=self._protocol,
+            load_args=self._load_args,
+            save_args=self._save_args,
+        )
+
+    def _exists(self) -> bool:
+        load_path = get_filepath_str(self._filepath, self._protocol)
+
+        if self._is_multifile:
+            files = self._fs.glob(load_path)
+            exists = True if files else False
+        else:
+            exists = self._fs.exists(load_path)
+
+        return exists
+
+    def _invalidate_cache(self):
+        """Invalidate underlying filesystem caches."""
+        filepath = get_filepath_str(self._filepath, self._protocol)
+        self._fs.invalidate_cache(filepath)
+
+    def __del__(self):
+        """Cleanup temporary directory"""
+        if self._temppath is not None:
+            logger.info("Deleting local temporary files.")
+            temp_filepath = str(self._temppath) + "/" + self._filepath.stem
+            if self._is_multifile:
+                temp_files = glob(temp_filepath)
+                for file in temp_files:
+                    try:
+                        os.remove(file)
+                    except FileNotFoundError:  # pragma: no cover
+                        pass  # pragma: no cover
+            else:
+                temp_filepath = temp_filepath + self._filepath.suffix
+                try:
+                    os.remove(temp_filepath)
```

Comment:
Minor, but we generally try to use [Pathlib](https://docs.python.org/3/library/pathlib.html) instead of `os` for file path handling in Kedro where possible as it's more robust across operating systems.

Reply:
Updated! I had trouble with `.unlink()` previously, but just dropped it in for `os.remove` and it looks good.